### PR TITLE
Add shallow context pass test

### PR DESCRIPTION
**Pull request by laskos**
Add simple test to check, that context from shallow render function is passing to component.

#3721

### CODE REVIEW

**Comment by laskos** [comment]
It's wild, that tests doesnt pass, becouse they fail at 

```
not ok 345 - renderers/dom/client/__tests__/ReactWebWorker-test ReactWebWorker can run React in a web worker.
```

My test is ok. Is it possible to rebuild?

**Comment by zpao** [comment]
The WebWorker test seems to have gone flaky again, nothing to do with you :)

@graue, you're most familiar with shallow testing, can you take a quick look?

File: src/test/__tests__/ReactTestUtils-test.js [hunk 1]
```diff
@@ -159,6 +159,24 @@ describe('ReactTestUtils', function() {
     expect(result).toEqual(<div />);
   });
 
+  it('can pass context with shallowly render', function() {
```
Comment (graue) [inline src/test/__tests__/ReactTestUtils-test.js]: nit: maybe call it `'can pass context when shallowly rendering'`?

File: src/test/__tests__/ReactTestUtils-test.js [hunk 2]
```diff
@@ -159,6 +159,24 @@ describe('ReactTestUtils', function() {
...
+        name: React.PropTypes.string,
+      },
+      render: function() {
+        return <div>{this.context.name}</div>;
+      },
+    });
+
+    var shallowRenderer = ReactTestUtils.createRenderer();
+    shallowRenderer.render(<SimpleComponent />, {
+        name: "foo"
```
Comment (graue) [inline src/test/__tests__/ReactTestUtils-test.js]: nit: add a trailing comma here

**Comment by graue** [comment]
Thanks, looks great! Just two minor issues.

**Comment by zpao** [comment]
Could you squash the commits and then we can merge this.

**Comment by laskos** [comment]
@zpao done

**Comment by zpao** [comment]
Thanks!

**Pull request merged**